lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -2993,6 +2996,29 @@ ThreadContext::InvalidatePropertyGuardEntry(const Js::PropertyRecord* propertyRe
             }
 
             guard->Invalidate();
+
+            // If a propertyGuard is invalidated, make sure to also remove it's entry from other unique property guard table of other property records.
+            if (performCleanupOfUniquePropertyGuards)
+            {
+                guards.Map([=](const Js::PropertyRecord* otherPropertyRecord, PropertyGuardEntry* guardEntry, const Memory::RecyclerWeakReference<const Js::PropertyRecord >* otherPropertyRecordWeakRef)
```

Comment:
This looks very expensive. Is there any perf impact? Can this be done only periodically?